### PR TITLE
[Dotenv] Add PHPDoc for `$overrideExistingVars`

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 5.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | -
| License       | MIT

Following https://github.com/symfony/symfony-docs/pull/19082#issuecomment-1778611355

### CODE REVIEW

File: src/Symfony/Component/Dotenv/Dotenv.php [hunk 1]
```diff
@@ -98,10 +98,11 @@ public function load(string $path, string ...$extraPaths): void
...
      *
-     * @param string      $path       A file to load
-     * @param string|null $envKey     The name of the env vars that defines the app env
-     * @param string      $defaultEnv The app env to use when none is defined
-     * @param array       $testEnvs   A list of app envs for which .env.local should be ignored
+     * @param string      $path                 A file to load
+     * @param string|null $envKey               The name of the env vars that defines the app env
+     * @param string      $defaultEnv           The app env to use when none is defined
+     * @param array       $testEnvs             A list of app envs for which .env.local should be ignored
+     * @param bool        $overrideExistingVars Whether existing environment variables set by the system must be overridden. Variables defined in .env files won't be overridden.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Dotenv/Dotenv.php]: I'm not sure about the last part of the sentence (+ we don't use final dots for these entries)
```suggestion
     * @param bool        $overrideExistingVars Whether existing environment variables set by the system must be overridden
```

File: src/Symfony/Component/Dotenv/Dotenv.php [hunk 2]
```diff
@@ -182,7 +183,7 @@ public function overload(string $path, string ...$extraPaths): void
      * Sets values as environment variables (via putenv, $_ENV, and $_SERVER).
      *
      * @param array $values               An array of env variables
-     * @param bool  $overrideExistingVars true when existing environment variables must be overridden
+     * @param bool  $overrideExistingVars Whether existing environment variables set by the system must be overridden. Variables defined in .env files won't be overridden.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Dotenv/Dotenv.php]:
```suggestion
     * @param bool  $overrideExistingVars Whether existing environment variables set by the system must be overridden
```

File: src/Symfony/Component/Dotenv/Dotenv.php (hunk 1 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/Dotenv/Dotenv.php]: I updated and added a link to the doc, now that's documented (like it's been done [here](https://github.com/symfony/symfony/pull/51974/files#diff-deee1a5892810656ac3eb9f0835a435cd95b01bc7bb79fccefde595c6428f992R37) for example)

File: src/Symfony/Component/Dotenv/Dotenv.php [hunk 3]
```diff
@@ -98,10 +98,11 @@ public function load(string $path, string ...$extraPaths): void
...
      *
-     * @param string      $path       A file to load
-     * @param string|null $envKey     The name of the env vars that defines the app env
-     * @param string      $defaultEnv The app env to use when none is defined
-     * @param array       $testEnvs   A list of app envs for which .env.local should be ignored
+     * @param string      $path                 A file to load
+     * @param string|null $envKey               The name of the env vars that defines the app env
+     * @param string      $defaultEnv           The app env to use when none is defined
+     * @param array       $testEnvs             A list of app envs for which .env.local should be ignored
+     * @param bool        $overrideExistingVars Whether existing environment variables set by the system must be overridden (@see https://symfony.com/doc/current/configuration.html#overriding-environment-variables-defined-by-the-system)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Dotenv/Dotenv.php]: we try to always use "should" instead of "must" ;) + no doc link.
I'm going to update while merging

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @alexandre-daubois.

File: src/Symfony/Component/Dotenv/Dotenv.php (hunk 3 above)
Comment (alexandre-daubois) [reply src/Symfony/Component/Dotenv/Dotenv.php]: Alright, I'll think of this in the next occasion. Thank you for taking care of this!

**Pull request merged**